python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -690,11 +689,17 @@ def _flush(self, objs):
             if not obj._dirty:
                 continue
 
+            if hasattr(obj, _reg_id_str):
+                obj_id = getattr(obj, _reg_id_str)
+                if obj_id not in self._loaded_ids:
+                    continue
+            else:
+                continue
```

Comment:
This implies that we're still getting non-loaded ids with `_dirty=True`? That sounds bad.